File: August/Week1/Day6/Submissions/findAllDups.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var findDuplicates = function(nums) {
    let duplicateFilter = [];
    
    for(let i of nums) {
        i = Math.abs(i);
        
        if(nums[i-1] > 0) {
            nums[i-1] *= -1;
        } else {
            duplicateFilter.push(i);
        }
    }
    
    return duplicateFilter;
};

File: August/Week1/Day6/Submissions/findAllDups.py
class Solution:
    def findDuplicates(self, nums: List[int]) -> List[int]:
        duplicateFilter = []
        for i in nums:
            i = abs(i)
            
            if nums[i-1] > 0:
                nums[i-1] *= -1
            else:
                duplicateFilter.append(i)
                
        return duplicateFilter